scripts/build-crds

```diff
@@ -1,10 +1,29 @@
-#! /bin/sh
+#! /bin/bash
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -q|--quiet)
+            QUIET=true
+            shift
+            ;;
+        *)
+            echo "Unknown option: $1"
+            exit 1
+            ;;
+    esac
+done
 
 cd $(dirname $0)/../
 
 # This will return non-zero until all of our objects in ./pkg/apis can generate valid crds.
 # allowDangerousTypes is needed for struct that use floats
-controller-gen crd:generateEmbeddedObjectMeta=true,allowDangerousTypes=false paths=./pkg/apis/... output:crd:dir=./pkg/crds/yaml/generated
+GEN_CMD="controller-gen crd:generateEmbeddedObjectMeta=true,allowDangerousTypes=false paths=./pkg/apis/... output:crd:dir=./pkg/crds/yaml/generated"
 
+# supress errors when running controller-gen since we have not fully migrated all of our CRDs to be compatible.
+if [ "$QUIET" = true ]; then
+    $GEN_CMD > /dev/null 2>&1
+else
+    $GEN_CMD
+fi
```

Comment:
The logic does not handle the case where the `controller-gen` command encounters a "real" error and fails to generate CRDs.  Honestly, I do not know if that will ever happen, but as a prevention, we should check the return value of the `controller-gen` command. 

for example:
```
if [$? != 0]; then
 	echo "something goes wrong when generating CRDs"
	exit 1
fi 
```

Reply:
This would make the script break whenever someone updates, creates or removes a CRD. Rancher has separate runtime checks and unit tests to verify that the desired CRDs are generated.

Reply:
OK, if that is the case, I agree we should rely on those checks and validations.